File: src/timer_state.ml
open Core

module Live_splits = struct
  type t = Duration.t option list
end

type t =
  | Idle
  (* splits * start time *)
  | Timing of Live_splits.t * Time_ns.t
  (* splits * start time * paused time *)
  | Paused of Live_splits.t * Time_ns.t * Time_ns.t
  (* completed splits * start time *)
  | Done of Live_splits.t * Time_ns.t

File: src/timer_state.rs
use std::time::{Duration, Instant};

use crate::split_file::SplitFile;

#[derive(Debug)]
pub enum TimerMode {
    Initial,
    Running { start_time: Instant },
    Paused { elapsed_at_pause: Duration },
    Finished { start_time: Instant },
}

pub struct TimerState {
    pub split_file: SplitFile,
    pub mode: TimerMode,
    pub splits: Vec<Option<Duration>>,
    pub anim_ref_time: Instant,
}
